allennlp/training/gradient_descent_trainer.py

```diff
@@ -349,6 +350,22 @@ def _pytorch_model(self):
             return self.model
         return self._ddp_wrapped_model.model
 
+    def clip_gradient(self):
+        """
+        Performs gradient clipping.
+        If the model is in mixed precision training, we would first unscale the gradient.
+        """
+        if self._grad_clipping is not None:
+            # 1. We have to unscale the gradient before clipping
+            if self._scaler is not None:
+                optimizer_state = self._scaler._per_optimizer_states[id(self.optimizer)]
+                # 2. The `unscale_` should't be performed more than once per optimizer per step call,
```

Comment:
```suggestion
                # 2. The `unscale_` shouldn't be performed more than once per optimizer per step call,
```

Reply:
Oh, that's my ignorance 😰. Is it correct now?